Source/WebCore/platform/graphics/gtk/GdkSkiaUtilities.cpp

```diff
@@ -73,6 +73,18 @@ RefPtr<cairo_surface_t> skiaImageToCairoSurface(SkImage& image)
 }
 #endif
 
+GRefPtr<GdkPixbuf> skiaImageToGdkPixbuf(SkImage& image)
+{
+    SkPixmap pixmap;
+    if (!image.peekPixels(&pixmap))
+        return { };
+
+    // FIXME: This only supports RGB{A,X}_8888, so we could convert the SkImage if it was in a different format.
+    return adoptGRef(gdk_pixbuf_new_from_data(pixmap.addr8(), GDK_COLORSPACE_RGB, !pixmap.isOpaque(), 8, pixmap.width(), pixmap.height(), pixmap.rowBytes(), [](guchar*, void *data) {
```

Comment:
...and then here you want `::addr32()` because that returns a 32-bit per pixel pointer (4 bytes per pixel, 8bits per color component). Sorry if I wasn't clear with this in my earlier comment.

```suggestion
    return adoptGRef(gdk_pixbuf_new_from_data(pixmap.addr32(), GDK_COLORSPACE_RGB, !pixmap.isOpaque(), 8, pixmap.width(), pixmap.height(), pixmap.rowBytes(), [](guchar*, void *data) {
```

Reply:
I'm not sure I understand, `addr8()` is very much correct here.

Reply:
The `addr8()` accessor is used for 8bpp (bits per pixel) images (grayscale or indexed color). The documentation in `SkPixmap.h` is clear about this:

```cpp
    /** Returns readable base pixel address. Result is addressable as unsigned 8-bit bytes.
        Will trigger an assert() if SkColorType is not kAlpha_8_SkColorType or
        kGray_8_SkColorType, and is built with SK_DEBUG defined.

        One byte corresponds to one pixel.

        @return  readable unsigned 8-bit pointer to pixels
    */
```

What we want here is the 32bpp (4 bytes per pixel, 1 byte for each of `RGB{A,X}`) accessor, which is `addr32()`:

```cpp
    /** Returns readable base pixel address. Result is addressable as unsigned 32-bit words.
        Will trigger an assert() if SkColorType is not kRGBA_8888_SkColorType or
        kBGRA_8888_SkColorType, and is built with SK_DEBUG defined.

        One word corresponds to one pixel.

        @return  readable unsigned 32-bit pointer to pixels
    */
```

Reply:
So the GdkPixbuf constructor doesn't support any format that isn't 8 bits.

However you can construct a GdkTexture/cairo_surface from it and then create a GdkPixbuf off that, so I took this approach.